bot.py

```diff
@@ -0,0 +1,300 @@
+    """Проверяет доступность переменных окружения для работы программы."""
+    logger.info('***Работает check_tokens.')
+    logger.debug(f'TELEGRAM_TOKEN - {TELEGRAM_TOKEN}.')
+    logger.debug(f'TELEGRAM_ADMIN_ID - {TELEGRAM_ADMIN_ID}.')
+    return all([TELEGRAM_TOKEN, TELEGRAM_ADMIN_ID])
+
+
+def last_check(update, context):
+    """Возвращает результаты последней проверки."""
+    logger.info('***Работает last_check.')
+    chat = update.effective_chat
+    name = update.message.chat.username
+    message = ''.join(CHECK_RESULT)
+    context.bot.send_message(
+        chat_id=chat.id,
+        text=message,
+        reply_markup=BUTTONS
+    )
+    logger.info(
+        f'Запрошены результаты последней проверки. Пользователь {name}.')
+
+
+def bot_settings(update, context):
+    """Возвращает список проверяемых сайтов и периодичность проверки."""
+    logger.info('***Работает bot_settings.')
+    chat = update.effective_chat
+    name = update.message.chat.username
+    part_name = name[:4]
+    endpoints_str = ''
+    for e in ENDPOINTS:
+        endpoints_str += e + '\n'
+    message = (
+        f'Список проверямых сайтов:\n{endpoints_str}'
+        + f'Периодичность проверки - 1 раз в {RETRY_TIME} ч.')
+    context.bot.send_message(
+        chat_id=chat.id,
+        text=message,
+        reply_markup=BUTTONS
+    )
+    logger.info(
+        f'Запрошены настройки бота. Часть имени пользователя {part_name}.')
+
+
+def subscribe(update, context):
+    """Подписка на рассылку сообщений от бота."""
+    logger.info('***Работает subscribe.')
+    chat = update.effective_chat
+    name = update.message.chat.username
+    tg_id = update.message.chat.id
+    logger.debug(f'tg_id - {tg_id}')
+    telegram_ids = get_subscribers_ids()
+    logger.debug(f'telegram_ids - {telegram_ids}.')
+    tg_setting = config['tg_setting']
+    new_tg_setting = ''
+    if tg_id in telegram_ids:
+        telegram_ids.remove(tg_id)
+        logger.debug('telegram_ids после удаления id - '
+                     + f'{telegram_ids}.')
+        for i in range(len(telegram_ids)):
+            if i != (len(telegram_ids) - 1):
+                new_tg_setting += (str(telegram_ids[i]) + ',')
+            else:
+                new_tg_setting += str(telegram_ids[i])
+        message = ('Вы отписались от рассылки сообщений от бота.')
+        logger.info(f'Пользователь {name} отписалися от рассылки.')
+    else:
+        telegram_ids.append(tg_id)
+        logger.debug('telegram_ids id после добавления - '
+                     + f'{telegram_ids}.')
+        for i in range(len(telegram_ids)):
+            if i != (len(telegram_ids) - 1):
+                new_tg_setting += (str(telegram_ids[i]) + ',')
+            else:
+                new_tg_setting += str(telegram_ids[i])
+        message = ('Вы подписались на рассылку сообщений от бота.')
+        logger.info(f'Пользователь {name} подписался на рассылку.')
+    tg_setting['subscription_tg_ids'] = new_tg_setting
```

Comment:
Судя по всему (https://stackoverflow.com/a/33270082/4906544) можно сделать так
```suggestion
    if tg_id in telegram_ids:
            telegram_ids.remove(tg_id)
            message = ('Вы отписались от рассылки сообщений от бота.')
            logger.info(f'Пользователь {name} отписывается от рассылки.')
    else:
            telegram_ids.append(tg_id)
            message = ('Вы подписались на рассылку сообщений от бота.')
            logger.info(f'Пользователь {name} подписывается на рассылку.')
    tg_setting['subscription_tg_ids'] = ".".join( str(id) for id in telegram_ids )
```

Reply:
Согласен. Скорректировал, но вместо разделителя точки указал запятую.